### PR TITLE
[FIX] updating citing link in warning 

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

This update will print the correct zenodo pyDARN link in the citing warning for plotting in pyDARN

**issue:** #129 

## Approval

**Number of approvals:** 2 - 1 test, 1 review

## Test

```python
import matplotlib.pyplot as plt 
import datetime as dt
import pydarn

#Read in fitACF file
stime = dt.datetime(2010, 3, 5, 4, 8)
grid_file = "20100305.n.grd"
grid_data = pydarn.SuperDARNRead(grid_file).read_grid()

#pydarn.Grid.plot_grid(grid_data, start_time=stime)
pydarn.Fan.plot_fov(6, stime)
plt.show()
```

```bash
/usr/lib/python3.6/site-packages/pydarn/exceptions/warning_formatting.py: 15: UserWarning: Please make sure to cite pyDARN in publications that use plots created by pyDARN using DOI: https://zenodo.org/record/4558130. Citing information for SuperDARN data is found at https://pydarn.readthedocs.io/en/master/user/citing/
```

This should print for all the plotting methods.

### CODE REVIEW

File: pydarn/__init__.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 from .exceptions import radar_exceptions
 from .exceptions.warning_formatting import standard_warning_format
 from .exceptions.warning_formatting import only_message_warning_format
+from .exceptions.warning_formatting import citting_warning
```
Comment (carleyjmartin) [inline pydarn/__init__.py:25]:
```suggestion
from .exceptions.warning_formatting import citing_warning
```

File: pydarn/exceptions/warning_formatting.py [hunk 2]
```diff
@@ -5,6 +5,18 @@
 #
 # Disclaimer: pyDARN is licensed under the GPL v 3.0 found in LICENSE
 #
+import warnings
+
+def citting_warning():
```
Comment (carleyjmartin) [inline pydarn/exceptions/warning_formatting.py:10]:
```suggestion
def citing_warning():
```

File: pydarn/plotting/acf.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 from pydarn import (plot_exceptions, SuperDARNRadars,
                     standard_warning_format, time2datetime,
-                    check_data_type)
+                    check_data_type, citting_warning)
```
Comment (carleyjmartin) [inline pydarn/plotting/acf.py:17]:
```suggestion
                    check_data_type, citing_warning)
```

File: pydarn/plotting/acf.py [hunk 4]
```diff
@@ -257,11 +257,7 @@ def plot_acfs(cls, dmap_data: List[dict], beam_num: int = 0,
                           date=time.strftime("%Y %b %d %H:%M"),
                           cpid=record['cp'])
         ax.set_title(title)
-        warnings.warn("Please make sure to cite pyDARN in publications that"
-                      " use plots created by pyDARN using DOI:"
-                      " https://zenodo.org/record/3978643. Citing information"
-                      " for SuperDARN data is found at"
-                      " https://pydarn.readthedocs.io/en/master/user/citing/")
+        citting_warning()
```
Comment (carleyjmartin) [inline pydarn/plotting/acf.py:260]:
```suggestion
        citing_warning()
```

File: pydarn/plotting/fan.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
 # Third party libraries
 import aacgmv2
 
-from pydarn import PyDARNColormaps, build_scan, radar_fov
+from pydarn import PyDARNColormaps, build_scan, radar_fov, citting_warning
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:30]:
```suggestion
from pydarn import PyDARNColormaps, build_scan, radar_fov, citing_warning
```

File: pydarn/plotting/fan.py [hunk 6]
```diff
@@ -221,7 +221,7 @@ def plot_fan(cls, dmap_data: List[dict], ax=None, scan_index: int = 1,
 
             if colorbar_label != '':
                 cb.set_label(colorbar_label)
-
+        citting_warning()
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:224]:
```suggestion
        citing_warning()
```

File: pydarn/plotting/fan.py [hunk 7]
```diff
@@ -324,7 +327,5 @@ def plot_fov(cls, stid: str, dtime: dt.datetime, ax=None,
             r = np.append(r, np.flip(rs[0:ranges[1], thetas.shape[1]-2]))
             r = np.append(r, np.flip(rs[0, 0:thetas.shape[1]-2]))
             ax.fill(theta, r, color=fov_color, alpha=alpha)
-            #ax.fill(thetas[ranges[1]-1, 0:thetas.shape[1]-1],
-            #        rs[ranges[1]-1, 0:thetas.shape[1]-1],
-            #        color=fov_color, alpha=alpha)
+        citting_warning()
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:330]:
```suggestion
        citing_warning()
```

File: pydarn/plotting/power.py [hunk 8]
```diff
@@ -6,7 +6,7 @@
 
 from typing import List
 
-from pydarn import SuperDARNRadars, exceptions, RTP
+from pydarn import SuperDARNRadars, exceptions, RTP, citting_warning
```
Comment (carleyjmartin) [inline pydarn/plotting/power.py:9]:
```suggestion
from pydarn import SuperDARNRadars, exceptions, RTP, citing_warning
```

File: pydarn/plotting/rtp.py [hunk 9]
```diff
@@ -17,7 +17,7 @@
 from pydarn import (gate2slant, check_data_type, time2datetime,
                     rtp_exceptions, plot_exceptions, SuperDARNCpids,
                     SuperDARNRadars, standard_warning_format,
-                    PyDARNColormaps)
+                    PyDARNColormaps, citting_warning)
```
Comment (carleyjmartin) [inline pydarn/plotting/rtp.py:20]:
```suggestion
                    PyDARNColormaps, citing_warning)
```

File: pydarn/plotting/rtp.py [hunk 10]
```diff
@@ -438,11 +439,7 @@ def plot_range_time(cls, dmap_data: List[dict], parameter: str = 'v',
                                                       norm) from None
         if colorbar_label != '':
             cb.set_label(colorbar_label)
-        warnings.warn("Please make sure to cite pyDARN in publications that"
-                      " use plots created by pyDARN using DOI:"
-                      " https://zenodo.org/record/3978643. Citing information"
-                      " for SuperDARN data is found at"
-                      " https://pydarn.readthedocs.io/en/master/user/citing/")
+        citting_warning()
```
Comment (carleyjmartin) [inline pydarn/plotting/rtp.py:442]:
```suggestion
        citing_warning()
```

File: pydarn/plotting/rtp.py [hunk 11]
```diff
@@ -674,11 +671,7 @@ def plot_time_series(cls, dmap_data: List[dict],
         ax.margins(x=0)
         ax.tick_params(axis='y', which='minor')
 
-        warnings.warn("Please make sure to cite pyDARN in publications that"
-                      " use plots created by pyDARN using DOI:"
-                      " https://zenodo.org/record/3978643. Citing information"
-                      " for SuperDARN data is found at"
-                      " https://pydarn.readthedocs.io/en/master/user/citing/")
+        citting_warning()
```
Comment (carleyjmartin) [inline pydarn/plotting/rtp.py:674]:
```suggestion
        citing_warning()
```

File: pydarn/plotting/rtp.py [hunk 12]
```diff
@@ -1035,12 +1028,7 @@ def plot_summary(cls, dmap_data: List[dict], beam_num: int = 0,
                      color='gray', ha='right', va='top',
                      rotation=-38, alpha=0.3)
 
-        warnings.warn("Please make sure to cite pyDARN in publications that"
-                      " use plots created by pyDARN using DOI:"
-                      " https://zenodo.org/record/3978643. Citing information"
-                      " for SuperDARN data is found at"
-                      " https://pydarn.readthedocs.io/en/master/user/citing/")
-
+        citting_warning()
```
Comment (carleyjmartin) [inline pydarn/plotting/rtp.py:1031]:
```suggestion
        citing_warning()
```

**Review by carleyjmartin — COMMENTED** [review]
L335 in power.py was spelled differently to how the function was defined, but that was the right way so I've made suggestions to make it easier to change so they're all the same, I think I got them all. 
I'll test first thing tomorrow. :D

**Comment by carleyjmartin** [comment]
Just checking, this has been added to the release milestone for 2.1, so we would have to update this before release anyway as the link is for v2.0.1?

Otherwise, no issues testing 👍

**Review by carleyjmartin — APPROVED** [review]

**Review by Devin-Huyghebaert — APPROVED** [review]
Tested.  Links in warning work as expected.

**Pull request merged**